Source/CoreExtensions/Private/CoreExtWaitForPrimaryPawn.cpp

```diff
@@ -0,0 +1,89 @@
+#include "CoreExtWaitForPrimaryPawn.h"
+
+#include "GameFramework/GameModeBase.h"
+
+#include <Engine/GameInstance.h>
+
+UCoreExtWaitForPrimaryPawn * UCoreExtWaitForPrimaryPawn::WaitForPrimaryPawn( UObject * world_context_object )
+{
+    ThisClass * action = nullptr;
+
+    if ( auto * world = GEngine->GetWorldFromContextObject( world_context_object, EGetWorldErrorMode::LogAndReturnNull ) )
+    {
+        action = NewObject< ThisClass >();
```

Comment:
Instead declare and define the var here with auto *
Return at the end of the scope
Return nullptr outside the scope

Reply:
Done